tests/test_diff_command.rs

```diff
@@ -313,3 +313,126 @@ fn test_color_words_diff_missing_newline() {
        9     : I
     "###);
 }
+
+#[test]
+fn test_diff_skipped_context() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "a\nb\nc\nd\ne\nf\ng\nh\ni\nj").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "=== Left side of diffs"]);
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+
+    std::fs::write(repo_path.join("file1"), "A\nb\nc\nd\ne\nf\ng\nh\ni\nJ").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "=== Must skip 2 lines"]);
```

Comment:
nit: You can combine the `describe` and `new` calls through (`jj new` accepts `-m`). If you don't like the asymmetry of using `describe` only for the first commit, just it's fine to just create an extra commit there too (i.e. `jj init repo --git; jj new @- -m '=== Left side of diffs'`(

Reply:
Simplified it a bit.

Reply:
Looks good. Feel free to merge this PR. You just need to rebase it first. (I stole the first spot on the list of contributors for this release :))